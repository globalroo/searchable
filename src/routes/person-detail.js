import React from "react";
import { getPersonById } from "src/tmdb-service/tmdb-api";
import { useByIdLoader } from "../hooks/by-id-loader";

export const PersonDetail = ({ id }) => {
	const { response } = useByIdLoader({ id, fetcher: getPersonById });
	return <div>{JSON.stringify(response)}</div>;
};
