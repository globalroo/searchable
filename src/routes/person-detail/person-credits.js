import React from "react";
import { getPersonCreditsById } from "src/tmdb-service/tmdb-api";
import { AssetScroller } from "src/components/asset-scroller";

import { useByIdLoader } from "../../hooks/by-id-loader";

export const PersonCredits = ({ id }) => {
	const { response } = useByIdLoader({ id, fetcher: getPersonCreditsById });
	const { payload } = response;

	if (!payload) return null;

	const { cast } = payload;

	return (
		<div className="content-row-2">
			<h1>Other Credits</h1>
			<AssetScroller assets={cast} />
		</div>
	);
};
