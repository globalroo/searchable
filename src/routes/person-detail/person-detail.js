import React from "react";
import { getPersonById, getPersonCreditsById } from "src/tmdb-service/tmdb-api";
import { getSmallPosterImage } from "src/tmdb-service/tmdb-config";
import { AssetScroller } from "src/components/asset-scroller";
import { useByIdLoader } from "../../hooks/by-id-loader";
import { SideBar } from "src/components/side-bar";
import { Age } from "src/components/age";
import { Typography, Divider, GridList } from "@material-ui/core";

const styles = {
	root: {
		display: "flex",
		flexWrap: "wrap",
		justifyContent: "space-between",
		overflow: "hidden"
	},
	gridList: {
		flexWrap: "nowrap",
		transform: "translateZ(0)",
		width: "300px"
	},
	gridListTile: {
		width: "110px"
	},
	spacer: { padding: "5px" }
};

const NaivePictureScroller = ({ profiles = [], name }) => {
	if (profiles.length === 0) return null;
	return (
		<GridList style={styles.gridList} component="div">
			{profiles.map((member, ix) => (
				<div key={`${member.id}_${ix}`}>
					<img
						src={getSmallPosterImage(member.file_path)}
						alt={name}
						style={styles.gridListTile}
					/>
					<Typography align="center">{member.name}</Typography>
					<Typography align="center" variant="caption">
						{member.character}
					</Typography>
				</div>
			))}
		</GridList>
	);
};
const OtherCredits = ({ id }) => {
	const { response } = useByIdLoader({ id, fetcher: getPersonCreditsById });
	const { payload } = response;

	if (!payload) return null;

	const { cast } = payload;

	return (
		<div className="content-row-2">
			<h1>Other Credits</h1>
			<AssetScroller assets={cast} />
		</div>
	);
};
export const PersonDetail = ({ id }) => {
	const { response, error, loading } = useByIdLoader({ id, fetcher: getPersonById });

	const { payload } = response;

	if (loading) return null;
	if (error) return <div>Error</div>;
	if (!payload) return null;

	const { birthday, name, biography, profile_path, images, deathday } = payload;

	const { profiles } = images;
	return (
		<div className="wrapper">
			<header className="header">
				{name} <Age birthday={birthday} deathday={deathday} />
			</header>

			<SideBar poster_path={profile_path} title={name} images={images.profiles}>
				<NaivePictureScroller profiles={profiles} name={name} />
			</SideBar>

			<article className="content">
				<h1 className="bold-header-uppercase">Biography</h1>
				<Divider />
				<br />
				<div className="content-row-1">
					<Typography>{biography}</Typography>
				</div>
				<br />
				<Divider />
				<br />
				<OtherCredits id={id} />
			</article>
		</div>
	);
};
