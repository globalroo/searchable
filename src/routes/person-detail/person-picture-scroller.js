import React from "react";
import { getSmallPosterImage } from "src/tmdb-service/tmdb-config";
import { GridList } from "@material-ui/core";
import { useImageLoader } from "src/hooks/image-loader";

const styles = {
	gridList: {
		flexWrap: "nowrap",
		transform: "translateZ(0)",
		width: "302px",
		paddingTop: "3px"
	},
	gridListTile: {
		width: "110px"
	},
	spacer: {
		padding: 2
	}
};
const PersonImage = ({ member, name, imgKey }) => {
	const [image] = useImageLoader(getSmallPosterImage(member[`${imgKey}`]));
	return (
		<div style={styles.spacer}>
			<img src={image} alt={name} className="undecorated-image" style={styles.gridListTile} />
		</div>
	);
};

export const PersonPictureScroller = ({ images = [], name, imgKey = "file_path" }) => {
	if (images.length <= 1) return null;
	return (
		<GridList style={styles.gridList} component="div">
			{images.map((member, ix) => (
				<PersonImage
					key={`${member.id}_${ix}`}
					member={member}
					ix={ix}
					name={name}
					imgKey="file_path"
				/>
			))}
		</GridList>
	);
};
