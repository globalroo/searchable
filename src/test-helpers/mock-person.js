export const mockPerson = {
	birthday: "1962-07-03",
	known_for_department: "Acting",
	deathday: null,
	id: 500,
	name: "Tom Cruise",
	images: {
		profiles: [
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 5,
				height: 1500,
				vote_average: 5.0350140056022,
				file_path: "/nPGebm4wxlLThpEEjTKthjoHMbM.jpg",
				width: 1000
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 6,
				height: 1500,
				vote_average: 5.1138716356108,
				file_path: "/hIVsySa8Ro6vrQ0hygIF6ylBgK7.jpg",
				width: 1000
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 8,
				height: 1500,
				vote_average: 5.1173708920188,
				file_path: "/fttvtGtgjPjoKEqGtnebbwEkVNA.jpg",
				width: 1000
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 22,
				height: 2100,
				vote_average: 5.282,
				file_path: "/3oWEuo0e8Nx8JvkqYCDec2iMY6K.jpg",
				width: 1400
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 7,
				height: 1050,
				vote_average: 5.1972789115646,
				file_path: "/hOoTazc5VZdXvMTLGpu9H7nb1e7.jpg",
				width: 700
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 11,
				height: 1800,
				vote_average: 5.302,
				file_path: "/54wOvUh3vWOTDqPMDq1VTYuh9yv.jpg",
				width: 1200
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 8,
				height: 1800,
				vote_average: 5.392,
				file_path: "/2Dkx4uuGoWFrPSitxdikv9z5azR.jpg",
				width: 1200
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 0,
				height: 600,
				vote_average: 0,
				file_path: "/s57wicK5nPuYhglUFUWECjnomJL.jpg",
				width: 400
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 0,
				height: 840,
				vote_average: 0,
				file_path: "/9pTIJwjH3Wg8JAxt0TNTpu0cXK0.jpg",
				width: 560
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 3,
				height: 600,
				vote_average: 5.318,
				file_path: "/gThaIXgpCm3PCiXwFNDBJCme85y.jpg",
				width: 400
			},
			{
				iso_639_1: null,
				aspect_ratio: 0.66666666666667,
				vote_count: 0,
				height: 1500,
				vote_average: 0,
				file_path: "/bn9CwMSM3o4XPZEPZdQ3grslGup.jpg",
				width: 1000
			}
		]
	},
	also_known_as: [
		"Том Круз",
		"トム・クルーズ",
		"ทอม ครูซ",
		"湯姆·克魯斯",
		"톰 크루즈",
		"توم كروز",
		"Thomas Cruise Mapother IV",
		"Τομ Κρουζ",
		"טום קרוז"
	],
	gender: 2,
	biography:
		'Thomas "Tom" Cruise (born Thomas Cruise Mapother IV; July 3, 1962) is an American actor and filmmaker. He has been nominated for three Academy Awards and has won three Golden Globe Awards.\n\nHe started his career at age 19 in the 1981 film Endless Love. After portraying supporting roles in Taps (1981) and The Outsiders (1983), his first leading role was in Risky Business, released in August 1983. Cruise became a full-fledged movie star after starring as Pete "Maverick" Mitchell in Top Gun (1986). He has since 1996 been well known for his role as secret agent Ethan Hunt in the Mission: Impossible film series. One of the biggest movie stars in Hollywood, Cruise has starred in many successful films, including The Color of Money (1986), Cocktail (1988), Rain Man (1988), Born on the Fourth of July (1989), Far and Away(1992), A Few Good Men (1992), The Firm (1993), Interview with the Vampire: The Vampire Chronicles (1994), Jerry Maguire (1996), Eyes Wide Shut (1999), Magnolia (1999), Vanilla Sky (2001), Minority Report (2002),The Last Samurai (2003), Collateral (2004), War of the Worlds (2005), Lions for Lambs (2007), Valkyrie (2008), Knight and Day (2010), Jack Reacher (2012), Oblivion (2013), and Edge of Tomorrow (2014).\n\nIn 2012, Cruise was Hollywood\'s highest-paid actor. Fifteen of his films grossed over $100 million domestically; twenty-one have grossed in excess of $200 million worldwide. Cruise is known for his support for the Church of Scientology and its affiliated social programs.',
	popularity: 14.983,
	place_of_birth: "Syracuse, New York, USA",
	profile_path: "/2Dkx4uuGoWFrPSitxdikv9z5azR.jpg",
	adult: false,
	imdb_id: "nm0000129",
	homepage: "http://www.tomcruise.com/"
};

export const mockCast = {
	cast: [
		{
			id: 74,
			character: "Ray Ferrier",
			original_title: "War of the Worlds",
			overview:
				"Ray Ferrier is a divorced dockworker and less-than-perfect father. Soon after his ex-wife and her new husband drop of his teenage son and young daughter for a rare weekend visit, a strange and powerful lightning storm touches down.",
			vote_count: 4208,
			video: false,
			media_type: "movie",
			poster_path: "/xXMM9KY2eq1SDOQif9zO91YOBA8.jpg",
			backdrop_path: "/bwTb66gicrB7HbkogwNs9V37Ddb.jpg",
			popularity: 16.077,
			title: "War of the Worlds",
			original_language: "en",
			genre_ids: [12, 878, 53],
			vote_average: 6.3,
			adult: false,
			release_date: "2005-06-29",
			credit_id: "52fe4213c3a36847f800226b"
		}
	]
};

export const mockCrew = {
	crew: [
		{
			id: 616,
			department: "Production",
			original_language: "en",
			original_title: "The Last Samurai",
			job: "Producer",
			overview:
				"Nathan Algren is an American hired to instruct the Japanese army in the ways of modern warfare, which finds him learning to respect the samurai and the honorable principles that rule them. Pressed to destroy the samurai's way of life in the name of modernization and open trade, Algren decides to become an ultimate warrior himself and to fight for their right to exist.",
			vote_count: 3267,
			video: false,
			media_type: "movie",
			release_date: "2003-12-05",
			vote_average: 7.4,
			title: "The Last Samurai",
			popularity: 15.985,
			genre_ids: [18, 28, 10752, 36],
			backdrop_path: "/lm15lu2jt3I3Syxsb90UJzUzMRV.jpg",
			adult: false,
			poster_path: "/sLv5pXysIz7QbtKFJy85d5yxv2W.jpg",
			credit_id: "52fe425ec3a36847f8018e1f"
		}
	]
};
