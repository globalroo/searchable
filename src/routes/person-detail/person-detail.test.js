import React from "react";
import { render } from "react-testing-library";

import { PersonDetail } from "./person-detail";

import { mockPerson, mockCast } from "src/test-helpers/mock-person";

let mockError = false;
let mockLoading = false;
let mockPayload = { ...mockPerson, ...mockCast };

jest.mock("src/hooks/by-id-loader", () => ({
	useByIdLoader: () => ({
		response: { payload: mockPayload },
		loading: mockLoading,
		error: mockError
	})
}));

describe("Person Detail tests", () => {
	afterEach(() => {
		mockLoading = false;
		mockError = false;
		mockPayload = { ...mockPerson, ...mockCast };
	});
	it("Renders a Person detail component", async () => {
		const { container } = await render(<PersonDetail id={mockPerson.id} />);
		expect(container).toMatchSnapshot();
	});
	it("Renders a Person detail component with no biography", async () => {
		mockPayload = { ...mockPayload, biography: undefined };
		const { container } = await render(<PersonDetail id={mockPerson.id} />);
		expect(container).toMatchSnapshot();
	});
	it("Renders a Person detail loading component", async () => {
		mockLoading = true;
		const { container } = await render(<PersonDetail id={mockPerson.id} />);
		expect(container).toMatchSnapshot();
	});
	it("Renders a Person detail error component", async () => {
		mockError = true;
		const { container } = await render(<PersonDetail id={mockPerson.id} />);
		expect(container).toMatchSnapshot();
	});
	it("Renders a Person detail error component", async () => {
		mockPayload = null;
		const { container } = await render(<PersonDetail id={mockPerson.id} />);
		expect(container).toMatchSnapshot();
	});
});
