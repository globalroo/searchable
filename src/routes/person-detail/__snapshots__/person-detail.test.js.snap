// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Detail tests Renders a Person detail component 1`] = `
<div>
  {"birthday":"1989-10-01","known_for_department":"Acting","deathday":null,"id":60073,"name":"Brie Larson","also_known_as":["Brianne Sidonie Desaulniers","بري لارسون","브리 라슨","ブリー・ラーソン","Бри Ларсон","บรี ลาร์สัน","布丽·拉尔森","Брі Ларсон"],"gender":1,"biography":"Brianne Sidonie Desaulniers (born October 1, 1989), known professionally as Brie Larson, is an American actress, director, and singer. Born in Sacramento, California, Larson was home-schooled before she studied acting at the American Conservatory Theater. She began her acting career in television, appearing as a regular on the 2001 sitcom Raising Dad, for which she was nominated for a Young Artist Award.\\n\\nAs a teenager, Larson had brief roles in the 2004 films 13 Going on 30 and Sleepover. Her performance in the comedy film Hoot (2006) was praised, and she subsequently played supporting roles in the films Greenberg (2010), Scott Pilgrim vs. the World (2010), 21 Jump Street (2012), and Don Jon (2013). From 2009 to 2011, Larson featured as a rebellious teenager in the television series United States of Tara.\\n\\nLarson's breakthrough role came with the independent drama Short Term 12 (2013), for which she received critical acclaim. Further success came in 2015 when she starred in Room, an acclaimed drama based on Emma Donoghue's novel of the same name. She won several awards for her portrayal of a troubled mother kidnap victim in the film, including the Academy Award, BAFTA Award, Critic's Choice Award, Golden Globe Award, Screen Actors Guild Award and Canadian Screen Award for Best Actress. In 2017, she starred as a war photographer in the adventure film Kong: Skull Island, her highest-grossing release.","popularity":41.58,"place_of_birth":"Sacramento, California, USA","profile_path":"/buGq7fC5iiLV5VvkdW4ui42k1yT.jpg","adult":false,"imdb_id":"nm0488953","homepage":null}
</div>
`;
